src/base-importer/BaseImporter.js

```diff
@@ -139,61 +356,122 @@ export default class BaseImporter {
-                    application: 2,
-                    password: 3
-                }
-                const dv = new DynamicValue(
-                    'com.luckymarmot.OAuth2DynamicValue',
-                    {
-                        grantType: grantMap[auth.get('flow')] || 0,
-                        authorizationUrl: auth.get('authorizationUrl') || '',
-                        accessTokenUrl: auth.get('tokenUrl') || '',
-                        scope: (auth.get('scopes') || []).join(' ')
-                    }
-                )
+        )
+    }
+
+    // @tested
+    _setOAuth1Auth(auth) {
+        return new DynamicValue(
+            'com.luckymarmot.OAuth1HeaderDynamicValue',
+            {
+                callback: auth.get('callback') || '',
+                consumerKey: auth.get('consumerKey') || '',
+                consumerSecret: auth.get('consumerSecret') || '',
+                tokenSecret: auth.get('tokenSecret') || '',
+                algorithm: auth.get('algorithm') || '',
+                nonce: auth.get('nonce') || '',
+                additionalParamaters: auth
+                    .get('additionalParamaters') || '',
+                timestamp: auth.get('timestamp') || '',
+                token: auth.get('token') || ''
+            }
+        )
+    }
+
+    // @tested
+    _setOAuth2Auth(auth) {
+        const grantMap = {
+            accessCode: 0,
+            implicit: 1,
+            application: 2,
+            password: 3
+        }
+        return new DynamicValue(
+            'com.luckymarmot.OAuth2DynamicValue',
+            {
+                grantType: grantMap[auth.get('flow')] || 0,
+                authorizationUrl: auth.get('authorizationUrl') || '',
+                accessTokenUrl: auth.get('tokenUrl') || '',
+                scope: (auth.get('scopes') || []).join(' ')
+            }
+        )
+    }
+
+    // @tested
+    _setAWSSig4Auth(auth) {
+        return new DynamicValue(
+            'com.shigeoka.PawExtensions.AWSSignature4DynamicValue',
+            {
+                key: auth.get('key') || '',
+                secret: auth.get('secret') || '',
+                region: auth.get('region') || '',
+                service: auth.get('service') || ''
+            }
+        )
+    }
+
+    // @tested
+    _setHawkAuth(auth) {
+        return new DynamicValue(
+            'uk.co.jalada.PawExtensions.HawkDynamicValue',
+            {
+                key: auth.get('key') || '',
+                id: auth.get('id') || '',
+                algorithm: auth.get('algorithm') || ''
+            }
+        )
+    }
+
+    // @tested
+    _setAuth(pawReq, auths) {
+        const authTypeMap = {
+            BasicAuth: this._setBasicAuth,
+            DigestAuth: this._setDigestAuth,
+            OAuth1Auth: this._setOAuth1Auth,
+            OAuth2Auth: this._setOAuth2Auth,
+            AWSSig4Auth: this._setAWSSig4Auth,
+            HawkAuth: this._setHawkAuth
+        }
```

Comment:
it doesn't matter in your case, but you always call these setAuth methods unbound… (it's fine since none of them use `this` inside), but maybe it would be better to refactor them to use this:

```javascript
_setBasicAuth = (auth) => {
  ...
}
```